stl/inc/flat_set

```diff
@@ -187,33 +208,36 @@ public:
         return _Get_cont().max_size();
     }
 
+    // modifiers
     template <class... _Args>
     auto emplace(_Args&&... _Vals) {
-        insert<_Kty>(_Kty{_STD forward<_Args>(_Vals)...});
+        constexpr bool _Is_key_type = _In_place_key_extract_set<_Kty, remove_cvref_t<_Args>...>::_Extractable;
+        if constexpr (_Is_key_type) {
+            return _Emplace(_STD forward<_Args>(_Vals)...);
+        } else {
+            return _Emplace(_Kty{_STD forward<_Args>(_Vals)...});

```

Comment:
Should this be `_Kty{_STD forward<_Args>(_Vals)...}` or `_Kty(_STD forward<_Args>(_Vals)...)`? (There are currenctly 4 occurrences of `_Kty{...}` in `emplace, emplace_hint, _Emplace and _Emplace_hint`)

Reply:
We should use `_Kty(_STD forward<_Args>(_Vals)...)` (direct-non-list-initialization).